src/modules/files-storage/api/controller/api-test/files-storage-download-upload.api.spec.ts

```diff
@@ -553,6 +553,156 @@ describe('files-storage controller (API)', () => {
 		});
 	});
 
+	describe('download multiple files action', () => {
+		describe('with not authenticated user', () => {
+			it('should return status 401', async () => {
+				const result = await testApiClient.post('/download', {
+					fileRecordIds: ['123'],
+					archiveName: 'test',
+				});
+
+				expect(result.status).toEqual(401);
+			});
+		});
+
+		describe('with bad request data', () => {
+			const setup = async () => {
+				jest.resetAllMocks();
+				const { studentUser, studentAccount } = UserAndAccountTestFactory.buildStudent();
+
+				const loggedInClient = testApiClient.loginByUser(studentAccount, studentUser);
+
+				const validId = new ObjectId().toHexString();
+
+				jest.spyOn(FileType, 'fileTypeStream').mockImplementation((readable) => Promise.resolve(readable));
```

Comment:
--> jest.spyOn(FileType, 'fileTypeStream').mockResolvedValueOnce(readable)

resolve und reject für Werte bei Promises
mockReturnValueOnce wenn kein Promise
wenn logic in der mock implementation verbaut wird, dann mockImplemenation

Immer mit Once arbeiten, zur Not Mehrfach.